### PR TITLE
Add support for String constants as parameter to @SuppressWarnings

### PR DESCRIPTION
**Pull request by mbring**
After my failed attempt to implement this last year (https://github.com/SonarSource/sonar-java/pull/585) I'm finally giving it another shot! I think I've covered all scenarios that I missed the last time but do let me know if something more needs to be done.

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java [hunk 1]
```diff
@@ -495,6 +495,36 @@ public String toString() {
   }
 
   /**
+   * Represents a final static field/variable.
+   */
+  public static class FinalStaticVariableJavaSymbol extends VariableJavaSymbol implements FinalStaticVariableSymbol {
+
+    VariableTree declaration;
```
Comment (sonartech-externalpr) [inline java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java]: BLOCKER "declaration" is the name of a field in "VariableJavaSymbol". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2387)

**Comment by sonartech-externalpr** [comment]
SonarQube analysis reported 2 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [SuppressWarningFilter.java#L168](https://github.com/SonarSource/sonar-java/blob/326ab66157f11014b19d0c055611fd4b242f6553/java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java#L168): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by benzonico** [comment]
Hi, thanks for the proposed implementation to tackle this issue. 

I created the following ticket as a consequence of your PR : https://jira.sonarsource.com/browse/SONARJAVA-2491 
https://jira.sonarsource.com/browse/SONARJAVA-2492

I won't follow through your implementation completely, mainly because I don't think that the design should add a method in a subinterface. Moreover, your suggested solution have a problem in the sense that it makes a difference if constant comes from bytecode or source. We tried to design semantic analysis API in such a way that it does not matter if something is read from bytecode or source code (so it simplifies writing of checks) and I think we can keep that principle in place while tackling this problem (as suggested in the created ticket). 

As a consequence of this, I'll close this PR and we will get (a bit more than just) inspiration from it to resolve your first problem which is accepting constants in annotations.
Hopefully this will get done in SonarJava 4.15. 
Thanks again for the contribution and implementation effort.

**Comment by mbring** [comment]
Great news! I don't care that you won't use my PR, I'm just happy you're planning to implement it. Thanks for the feedback!

Looking forward for SonarJava 4.15!

**Pull request closed without merging**